lib/utils.pm

```diff
@@ -327,8 +327,21 @@ sub unlock_if_encrypted {
         assert_screen("encrypted-disk-password-prompt", 200);
         type_password;    # enter PW at boot
         save_screenshot;
-        assert_screen 'encrypted_disk-typed_password' if $args{check_typed_password};
-        send_key "ret";
+        if ($args{check_typed_password}) {
+            if (check_screen "encrypted_disk-typed_password", 30) {
+                send_key "ret";
```

Comment:
What's about moving this outside of all of these conditions? We need to press enter in any of the cases, and we need retry only in case we failed to type it.

Reply:
The existing logic about checking typed password is to perform the check before pressing enter. If we move send_key "ret" before the conditions, we need to make sure that the check_screen waits enough for either the password to be asked again , in case of wrong password, or that the next screen is present, which I am not sure if it will be gnome login screen or something else. I don't know if this would be less error prone. Or did you have something else in mind?

Reply:
I can see that we have `send_key "ret";` in every code path, so I believe we can move it out and press unconditionally. Which will requites slight changes to the current code as we won't have anything special to do in case of `if (check_screen "encrypted_disk-typed_password", 30)`. Let me know if it's clear, otherwise I will provide code snippet for the proposal.

Reply:
```perl
if ($args{check_typed_password}) {
     if (!check_screen "encrypted_disk-typed_password", 30) {
            record_info("Invalid password", "Not all password characters were typed successfully, retyping");
            send_key "backspace" for (0 .. 9);
            type_password;
            assert_screen "encrypted_disk-typed_password";
            save_screenshot;
      }
}
send_key "ret";
```

Reply:
Ok, for some reason my brain interpreted "outside" as "before". Done, running VRs

Reply:
Did you forget to push to the branch?

Reply:
yes.... :'(

Reply:
No worries at all ;)